R/widerRepeated.R

```diff
@@ -0,0 +1,50 @@
+#' @name widerRepeated
+#' @title Transform Data Into Wide Format
+#' 
+#' @description Converts a dataframe into wide format given a single REDCap form.
+#' This function assumes that the Records argument is the result of exportRecordsTyped,
+#' and that all empty values have been previously dropped.
+#'   
+#' @param Records A \code{data.frame} containing the records from
+#'        \code{\link{exportRecordsTyped}}
+#' @param rcon A REDCap connection object as created by \code{unlockREDCap}.
+#' @export
+widerRepeated <- function(Records, rcon)
+{
+  idvar <- getProjectIdFields(rcon)
+  ###########################################################################
+  # Check parameters passed to function
+  coll <- checkmate::makeAssertCollection()
+  
+  checkmate::assert_data_frame(x = Records)
+  checkmate::assert_class(x = rcon,
+                          classes = "redcapApiConnection",
+                          add = coll)
+  checkmate::assert_character(x = idvar,
+                              len = 1,
+                              add = coll)
+  checkmate::assert_subset(x = idvar, names(Records))
+  checkmate::reportAssertions(coll)
+  
```

Comment:
We haven't really worked with data access groups yet, but I think that should be included among the variables that are considered ID variables--that is, I don't think we would want `redcap_data_access_group.1` and `redcap_data_access_group.2`. 

Should we consider adding:
```
  form_ids <- c(idvar, "redcap_data_access_group", 
                "redcap_event_name", "redcap_repeat_instrument")
  form_ids <- form_ids[form_ids %in% names(Records)]
```